Automattic/jetpack
# My Jetpack: Stats product card - various improvements

This PR makes various improvements to the My Jetpack "Stats" product card.  See 'Proposed changes' below:
and see project task: https://github.com/Automattic/jetpack-marketing/issues/505




## Proposed changes:

* Make large Stats card show by default (if Jetpack plugin activated) & remove small stats card.
* When site hasn't established first site connection, show "Learn more" CTA.
* When connected, allow user to view detailed stats, as well as upgrade/purchase a Stats plan. (two buttons)
* Display either "commmercial" interstitial or "pwyw" interstitial depending on if site "is commercial" and/or if site has a purchased Stats plan.

**Still TODO: (in a separate follow-up PR)**
- Show Stats card when stand-alone plugin(s) activated (Jetpack plugin not installed), and add relevant action button labels/behaviors

### Other information:


## Jetpack product discussion


Fixes: https://github.com/Automattic/jetpack/issues/35035
Related to: https://github.com/Automattic/jetpack-marketing/issues/505

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






- Spin up a jurassic.ninja site with Jetpack Beta plugin pointing to this branch (`update/my-jetpack-stats-section`).
- While Jetpack not connected yet, go to My Jetpack page.
- Verify you see the large Stats card at the top. Status: "inactive", CTA button label: "Learn more".  CTA action: `admin.php?page=my-jetpack#/add-stats`
- Click the "Learn more" button. Verify the Stats interstitial CTA reads, "Get Stats" (not "Install Stats").
- Click the ⬅️ Back button, to go back to My Jetpack.
- Next, connect Jetpack, but do not "Authorize".  (Site-only connection)
- Go back to the My Jetpack page.
- Examine the large Stats card. Verify the Stats card status is now "Active" and the primary action button reads, "Upgrade", and secondary button reads, "See detailed stats".
- Verify the "Upgrade button points to `?page=stats#!/stats/purchase/:blogId` and verify "See detailed stats" points to `?page=stats`.
- Next connect user account (Establish user-connection).
- Go back to the My Jetpack page. Verify the Stats card status is "Active" and the primary action button reads, "Upgrade", and secondary button reads, "See detailed stats".
- Go to Jetpack --> Settings, scroll down to footer, click the "Modules" link.
- Search "Stats", then click the "Deactivate" button for Jetpack Stats.
- Navigate back to My Jetpack page, and verify the Stats action button reads "Activate" and the Status is "Module disabled"
- In the Stats card, click the "Activate" button and verify the Stats module gets activated and the Stats card status changes to "Active".
    - Note: I think there is a bug on the Activate button where, on click of the button we should be preventing the browser click event default action, ie.- `event.preventDefault()`.  Will tackle this in a separate PR.
    (You may need to refresh/reload the page, maybe not).
- Now disconnect Jetpack (from the "Manage connection" button, either on Jetpack Dashboard or My Jetpack).
- Go to My Jetpack page. Verify the Stats CTA reads, "Connect Jetpack to use Stats" and the status is "Needs connection".
- Click the Connect button. Connect Jetpack (user connection).
- Click the "Upgrade" button and purchase the Stats PWYW plan.
- After purchased, verify the "Upgrade" button is no longer available.
- In Sore Admin, remove the PWYW plan you just purchased.
- Next, purchase Stats Commercial (any tier..., except **_not_** the highest tier (more than 1m views))
- The "Upgrade" button should still be visible.
- Now purchase the Stats commercial, the highest tier (more than 1m views).
- Verify the "Upgrade" button is no longer available when you own the highest tier of Stats commercial.
